frontend/src/store/controller/reducers.ts

```diff
@@ -61,9 +65,26 @@ const messageReducer = <T extends PBMessage>(
   }
 };
 
+const mergeEventList = (newEvents: NextLogEvents, oldEvents: NextLogEvents): NextLogEvents => {
+  const events = [...oldEvents.logEvents];
+  if (!oldEvents || !oldEvents.logEvents.length) {
+    return newEvents as NextLogEvents;
+  }
+  newEvents.logEvents.forEach((logEvent: LogEvent) => {
+    const id = events.find((ev: LogEvent) => ev.id === logEvent.id);
+    if (id === undefined) {
+      events.push(logEvent);
+    }
+  });
```

Comment:
There's definitely a more efficient way to merge two sorted arrays (`newEvents` and `oldEvents` should both have been sorted), but this implementation is fine for this PR.

Reply:
Has been fixed on #190